File: server/src/entity/Vehicle.cpp
//
// Created by zhou_zhengming on 2025/6/5.
//

#include "entity/Vehicle.h"

void Vehicle::updateOrder() {
    order->setCost(totalFee);
    order->setStart(start);
    order->setEnd(end);
    order->setType(mode);
}

void to_json(nlohmann::json& j, const Vehicle& v){
    j["type"] = v.mode;
    j["start"] = v.start;
    j["end"] = v.end;
    j["chargeTime"] = v.chargeTime;
    j["servFee"] = v.servFee;
    j["elecFee"] = v.elecFee;
    j["totalFee"] = v.totalFee;
    j["queueId"] = v.queueId;
    j["reqPower"] = v.reqPower;
}

File: client/charge-app.Core/Models/Vehicle.cs
using charge_app.Core.Helpers;
using Newtonsoft.Json;

namespace charge_app.Core.Models;

public class Vehicle
{
    public ChargingPile.ChargingPileType Type { get; set; }
    public uint QueueId { get; set; }
    public double ReqPower { get; set; }
    public double ChargeTime { get; set; }
    public long Start { get; set; }
    public long End { get; set; }
    public double ElecFee { get; set; }
    public double ServFee { get; set; }
    public double TotalFee { get; set; }

    [JsonIgnore] public string TextType => Type == ChargingPile.ChargingPileType.Fast ? "快充" : "慢充";
    [JsonIgnore] public string TextElecFee => $"{ElecFee:F2}元";
    [JsonIgnore] public string TextServFee => $"{ServFee:F2}元";
    [JsonIgnore] public string TextTotalFee => $"{TotalFee:F2}元";
    [JsonIgnore] public string TextChargeTime => $"{ChargeTime:F2}小时";
    [JsonIgnore] public string TextReqPower => $"{ReqPower:F2}度";
    [JsonIgnore] public string TextStart => TimeHelper.ConvertTime(Start);
    [JsonIgnore] public string TextEnd => TimeHelper.ConvertTime(End);


}